operator/builtin/parser/trace/trace_test.go

```diff
@@ -0,0 +1,155 @@
+	_, err := traceParserConfig.Build(testutil.NewBuildContext(t))
+	require.NoError(t, err)
+}
+
+func TestTraceParserParse(t *testing.T) {
+	cases := []struct {
+		name           string
+		inputRecord    map[string]interface{}
+		expectedRecord map[string]interface{}
+		expectErr      bool
+		traceId        string
+		spanId         string
+		traceFlags     string
+	}{
+		{
+			"AllFields",
+			map[string]interface{}{
+				"trace_id":    "480140f3d770a5ae32f0a22b6a812cff",
+				"span_id":     "92c3792d54ba94f3",
+				"trace_flags": "01",
+			},
+			map[string]interface{}{},
+			false,
+			"480140f3d770a5ae32f0a22b6a812cff",
+			"92c3792d54ba94f3",
+			"01",
+		},
+		{
+			"WrongFields",
+			map[string]interface{}{
+				"traceId":    "480140f3d770a5ae32f0a22b6a812cff",
+				"traceFlags": "01",
+				"spanId":     "92c3792d54ba94f3",
+			},
+			map[string]interface{}{
+				"traceId":    "480140f3d770a5ae32f0a22b6a812cff",
+				"spanId":     "92c3792d54ba94f3",
+				"traceFlags": "01",
+			},
+			false,
+			"",
+			"",
+			"",
+		},
+		{
+			"OnlyTraceId",
+			map[string]interface{}{
+				"trace_id": "480140f3d770a5ae32f0a22b6a812cff",
+			},
+			map[string]interface{}{},
+			false,
+			"480140f3d770a5ae32f0a22b6a812cff",
+			"",
+			"",
+		},
+		{
+			"WrongTraceIdFormat",
+			map[string]interface{}{
+				"trace_id":    "foo_bar",
+				"span_id":     "92c3792d54ba94f3",
+				"trace_flags": "01",
+			},
+			map[string]interface{}{},
+			true,
+			"",
+			"92c3792d54ba94f3",
+			"01",
+		},
+		{
+			"WrongTraceFlagFormat",
+			map[string]interface{}{
+				"trace_id":    "480140f3d770a5ae32f0a22b6a812cff",
+				"span_id":     "92c3792d54ba94f3",
+				"trace_flags": "foo_bar",
+			},
+			map[string]interface{}{},
+			true,
+			"480140f3d770a5ae32f0a22b6a812cff",
+			"92c3792d54ba94f3",
+			"",
+		},
+		{
+			"AllFields",
+			map[string]interface{}{
+				"trace_id":    "480140f3d770a5ae32f0a22b6a812cff",
+				"span_id":     "92c3792d54ba94f3",
+				"trace_flags": "01",
+			},
+			map[string]interface{}{},
+			false,
+			"480140f3d770a5ae32f0a22b6a812cff",
+			"92c3792d54ba94f3",
+			"01",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			traceParserConfig := NewTraceParserConfig("")
+			_, _ = traceParserConfig.Build(testutil.NewBuildContext(t))
+			e := entry.New()
+			e.Record = tc.inputRecord
+			err := traceParserConfig.Parse(e)
+			if tc.expectErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+			require.Equal(t, tc.expectedRecord, e.Record)
+			traceId, _ := hex.DecodeString(tc.traceId)
+			if len(tc.traceId) == 0 {
+				traceId = nil
+			}
+			spanId, _ := hex.DecodeString(tc.spanId)
+			if len(tc.spanId) == 0 {
+				spanId = nil
+			}
+			traceFlags, _ := hex.DecodeString(tc.traceFlags)
+			if len(tc.traceFlags) == 0 {
+				traceFlags = nil
+			}
+			require.Equal(t, traceId, e.TraceId)
+			require.Equal(t, spanId, e.SpanId)
+			require.Equal(t, traceFlags, e.TraceFlags)
```

Comment:
I don't understand why this is necessary. Is it really?

Reply:
Reworked a bit to make it more clear. The DecodeString is to make it easier to build the tests